### ISSUE TITLE
Cython warnings during installation

### ISSUE THREAD

**Issue by DevarshiChoudhury** [xpsi-group/xpsi#407]
The following warnings were generated during installation on a Surf Research Cloud (SRC) Ubuntu v22.04 system while installing X-PSI. I used the `basic_environment.yml` file to create my environment and as such I use Cython v0.29.37 (checked via `pip list`). 

These warnings didn't show up the last time I installed X-PSI in another SRC Ubuntu although using v20.04. Now the last time, I didn't separately captured the full `stderr` message, only the ouput and then the errors/warnings were printed in terminal. This time in Ubuntu v22.04, I captured them using `CC=gcc python setup.py install >out.txt 2>err.txt` and noticed these warnings. 

They do appear concerning since they are talking about issues surrounding truth validity of `and` statements. Has anyone else experienced this before?

```
xpsi/cellmesh/mesh.c: In function '__Pyx_Import':
xpsi/cellmesh/mesh.c:23317:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.mesh" + 4' must not be NULL [-Waddress]
23317 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/polar_mesh.c: In function '__Pyx_Import':
xpsi/cellmesh/polar_mesh.c:23574:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.polar_mesh" + 4' must not be NULL [-Waddress]
23574 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/global_mesh.c: In function '__Pyx_Import':
xpsi/cellmesh/global_mesh.c:21900:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.global_mesh" + 4' must not be NULL [-Waddress]
21900 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/rays.c: In function '__Pyx_Import':
xpsi/cellmesh/rays.c:24301:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.rays" + 4' must not be NULL [-Waddress]
24301 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/energy_interpolator.c: In function '__Pyx_Import':
xpsi/tools/energy_interpolator.c:19312:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.energy_interpolator" + 4' must not be NULL [-Waddress]
19312 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/energy_integrator.c: In function '__Pyx_Import':
xpsi/tools/energy_integrator.c:19223:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.energy_integrator" + 4' must not be NULL [-Waddress]
19223 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/phase_integrator.c: In function '__Pyx_Import':
xpsi/tools/phase_integrator.c:19126:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.phase_integrator" + 4' must not be NULL [-Waddress]
19126 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/phase_interpolator.c: In function '__Pyx_Import':
xpsi/tools/phase_interpolator.c:20487:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.phase_interpolator" + 4' must not be NULL [-Waddress]
20487 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/synthesise.c: In function '__Pyx_Import':
xpsi/tools/synthesise.c:23496:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.synthesise" + 4' must not be NULL [-Waddress]
23496 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/tools/__init__.c: In function '__Pyx_Import':
xpsi/tools/__init__.c:5323:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.tools.__init__" + 4' must not be NULL [-Waddress]
 5323 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/likelihoods/default_background_marginalisation.c: In function '__Pyx_Import':
xpsi/likelihoods/default_background_marginalisation.c:24235:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.likelihoods.default_background_marginalisation" + 4' must not be NULL [-Waddress]
24235 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/likelihoods/_poisson_likelihood_given_background.c: In function '__Pyx_Import':
xpsi/likelihoods/_poisson_likelihood_given_background.c:21452:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.likelihoods._poisson_likelihood_given_background" + 4' must not be NULL [-Waddress]
21452 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/likelihoods/_poisson_likelihood_given_background_IQU.c: In function '__Pyx_Import':
xpsi/likelihoods/_poisson_likelihood_given_background_IQU.c:21465:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.likelihoods._poisson_likelihood_given_background_IQU" + 4' must not be NULL [-Waddress]
21465 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/likelihoods/_gaussian_likelihood_given_background_IQU.c: In function '__Pyx_Import':
xpsi/likelihoods/_gaussian_likelihood_given_background_IQU.c:21434:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.likelihoods._gaussian_likelihood_given_background_IQU" + 4' must not be NULL [-Waddress]
21434 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/likelihoods/_gaussian_likelihood_QnUn.c: In function '__Pyx_Import':
xpsi/likelihoods/_gaussian_likelihood_QnUn.c:20298:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.likelihoods._gaussian_likelihood_QnUn" + 4' must not be NULL [-Waddress]
20298 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/__init__.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/__init__.c:22959:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.__init__" + 4' must not be NULL [-Waddress]
22959 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/preload.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/preload.c:18743:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.preload" + 4' must not be NULL [-Waddress]
18743 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_user.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_user.c:1997:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_user" + 4' must not be NULL [-Waddress]
 1997 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_BB.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_BB.c:1962:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_BB" + 4' must not be NULL [-Waddress]
 1962 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_Num4D.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_Num4D.c:3839:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_Num4D" + 4' must not be NULL [-Waddress]
 3839 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_BB_burst.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_BB_burst.c:1988:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_BB_burst" + 4' must not be NULL [-Waddress]
 1988 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_Num2D.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_Num2D.c:3726:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_Num2D" + 4' must not be NULL [-Waddress]
 3726 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_Num2D_split.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_Num2D_split.c:3698:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_Num2D_split" + 4' must not be NULL [-Waddress]
 3698 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/hot_Num5D_split.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/hot_Num5D_split.c:7554:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.hot_Num5D_split" + 4' must not be NULL [-Waddress]
 7554 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/surface_radiation_field/elsewhere_user.c: In function '__Pyx_Import':
xpsi/surface_radiation_field/elsewhere_user.c:1962:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.surface_radiation_field.elsewhere_user" + 4' must not be NULL [-Waddress]
 1962 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integrator.c: In function '__Pyx_Import':
xpsi/cellmesh/integrator.c:26527:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integrator" + 4' must not be NULL [-Waddress]
26527 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integratorIQU.c: In function '__Pyx_Import':
xpsi/cellmesh/integratorIQU.c:27358:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integratorIQU" + 4' must not be NULL [-Waddress]
27358 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integrator_for_azimuthal_invariance.c: In function '__Pyx_Import':
xpsi/cellmesh/integrator_for_azimuthal_invariance.c:26201:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integrator_for_azimuthal_invariance" + 4' must not be NULL [-Waddress]
26201 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integrator_for_azimuthal_invariance_split.c: In function '__Pyx_Import':
xpsi/cellmesh/integrator_for_azimuthal_invariance_split.c:26241:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integrator_for_azimuthal_invariance_split" + 4' must not be NULL [-Waddress]
26241 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integratorIQU_for_azimuthal_invariance.c: In function '__Pyx_Import':
xpsi/cellmesh/integratorIQU_for_azimuthal_invariance.c:27598:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integratorIQU_for_azimuthal_invariance" + 4' must not be NULL [-Waddress]
27598 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integratorIQU_for_azimuthal_invariance_split.c: In function '__Pyx_Import':
xpsi/cellmesh/integratorIQU_for_azimuthal_invariance_split.c:27749:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integratorIQU_for_azimuthal_invariance_split" + 4' must not be NULL [-Waddress]
27749 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/cellmesh/integrator_for_time_invariance.c: In function '__Pyx_Import':
xpsi/cellmesh/integrator_for_time_invariance.c:23705:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.cellmesh.integrator_for_time_invariance" + 4' must not be NULL [-Waddress]
23705 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
xpsi/pixelmesh/integrator.c: In function '__Pyx_Import':
xpsi/pixelmesh/integrator.c:23223:21: warning: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"xpsi.pixelmesh.integrator" + 4' must not be NULL [-Waddress]
23223 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                     ^~
```

**Comment by thjsal** [comment]
I am using the same Ubuntu and Cython versions, but for some reason I do not get these warnings.

**Comment by drannawatts** [comment]
I got a bunch of what appear to be Cython related warnings when I just installed on my Mac as well (relating to CYTHON_FALLTHROUGH).  Maybe the Cython 3 upgrade though will fix this?

**Comment by DevarshiChoudhury** [comment]
@drannawatts yes, I keep getting these too. I'm not sure what this related to

**Comment by evertrol** [comment]
This will depend on the gcc version: newer versions of compilers often have more (better) warnings and stricter checking. Why DevarshiChoudhury has a (presumably newer) noisier gcc than thjsal with the same OS, I don't know, but it sounds like you're not using the same gcc version. Possibly because SURF upgraded gcc on their cloud, or you loaded a module/environment with a more recent gcc.

This come from Cython itself (the generated C code). The Cython issue [is here](https://github.com/cython/cython/issues/5663), indicating this is the case for gcc 12.0 (and possibly later versions). It should be fixed in Cython 3.0.3.

This is not something that can be fixed on the X-PSI side, since it's a Cython issue.
Also, since the same code was likely generated without the warnings (and ran fine), it's unlikely to be a cause of concern in this particular case, and can be regarded as "just a warning".

**Comment by DevarshiChoudhury** [comment]
@evertrol so indeed these warnings in the original post are now gone upon using Cython v3.0.11.
